Custom signs

pdavydov108 opened the pull request:
This pull request adds a way for users to customize how error, warning, etc signs look on the screen. Four variables a provided for this. They should be set as a dict where keys are the names of parameters accepted by vim's `sign` command. This allows to not only set nice unicode sings for errors, but also to use highlight groups provided by custom colorschemes. This makes signs look much better, especially with those that support true color, such as gruvbox. See also #114.

prabirshrestha commented on README.md:
````diff
@@ -82,11 +82,20 @@ Plug 'prabirshrestha/asyncomplete-lsp.vim'
 ```
 
+This plugin defines four groups of signs: LspError, LspWarning, LspInformation, LspHint. It is possible to customize how this signs are printed on the screen by setting the following global variables:
+`g:lsp_signs_error`, `g:lsp_signs_warning`, `g:lsp_signs_information`, `g:lsp_signs_hint`. They should be set to a dict, where keys are the parameters supported by vim sign command (see `:help sign`). For example:
````
We should only allow users to modify text and not the texthl or linehl. We can provide LspError and users can modify it. We can default to the colorscheme errors similar to how Ale does.

pdavydov108 replied:
Okay, I can implement it this way, but then it will not be possible to set icon for a sign. Did I get you right that you want group links for texthl and linkhl?

prabirshrestha replied:
So what if we do this instead. 

```vim
let g:lsp_signs_error = {'text': '✗'}
let g:lsp_signs_hint = {'icon': '/path/to/some/icon'}
```

We can also support both at the same time. So in gui it defaults to icon and in terminal it uses text.

```vim
let g:lsp_signs_hint = {'icon': '/path/to/some/icon', 'text': '✗'}
```

But not allow people to modify `texthl` or `linehl`.

prabirshrestha commented on README.md:
````diff
@@ -82,11 +82,20 @@ Plug 'prabirshrestha/asyncomplete-lsp.vim'
+```viml
+let g:lsp_signs_error = {'text': '✗', 'texthl': 'GruvboxRedSign', 'linehl': 'Error'}
+let g:lsp_signs_warning = {'text': '‼', 'texthl': 'GruvboxYellowSign'}
+let g:lsp_signs_hint = {'info': '/path/to/some/icon'} " icons require GUI
````
did u mean `icon` instead of `info`

pdavydov108 replied:
Oh, sorry, of course, it's a typo.

pdavydov108 commented:
@prabirshrestha I've removed option to set texthl and linehl via dict. Also, I've added highlight groups for all signs, for texthl and for linehl, the same way as [ale does this](https://github.com/w0rp/ale/blob/92e6e4d1ba482a4d2d89d850f660c67ccf8a28eb/autoload/ale/sign.vim#L5). What do you think?

prabirshrestha commented:
@pdavydov108 Thanks again. Looks good.

dataO1 commented:
This is not documented in the docs. Ive searched a long time, since i found this, maybe someone can add it, so that future users find this more easily?
